test/functional/cmdLineTests/modularityddrtests/modularityddrtests.xml

```diff
@@ -630,4 +631,304 @@
 		<output regex="no" type="required">!findmodules package</output>
 		<output regex="no" type="failure">DDRInteractiveCommandException</output>
 	</test>
+
+	<!-- Test !dumpmodule and all options-->
+	<test id="Run !dumpmodule">
+		<exec command="sh" capture="LOGNAME" platforms="zos.*" >
+			<arg>-c</arg>
+			<arg>echo $$LOGNAME</arg>
+		</exec>
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP.X001' $DUMPFILE$" platforms="zos_390-64.*" />
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP' $DUMPFILE$" platforms="zos_390-31.*" />
+		<exec command="tso delete J9CORE.DMP.*" platforms="zos_390-64.*" />
+		<exec command="tso delete J9CORE.DMP" platforms="zos_390-31.*" />
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core</arg>
+			<arg>$DUMPFILE$</arg>
+			<input>!dumpmodule $javaBaseModuleAddr$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success">Module:</output>
+		<output regex="no" type="required">Exports:</output>
+		<output regex="no" type="required">Requires:</output>
+		<output regex="no" type="required">Exports java/nio/file/spi</output>
+		<output regex="no" type="required">!j9package 0x</output>
+		<output regex="no" type="required">Exports sun/util/logging</output>
+		<output regex="no" type="required">to java.logging</output>
+		<output regex="no" type="required">!j9module 0x</output>
+		<output regex="no" type="required">to ALL-UNNAMED</output>
+		<output regex="no" type="failure">DDRInteractiveCommandException</output>
+	</test>
+
+	<test id="Run !dumpmodule packages">
+		<exec command="sh" capture="LOGNAME" platforms="zos.*" >
+			<arg>-c</arg>
+			<arg>echo $$LOGNAME</arg>
+		</exec>
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP.X001' $DUMPFILE$" platforms="zos_390-64.*" />
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP' $DUMPFILE$" platforms="zos_390-31.*" />
+		<exec command="tso delete J9CORE.DMP.*" platforms="zos_390-64.*" />
+		<exec command="tso delete J9CORE.DMP" platforms="zos_390-31.*" />
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core</arg>
+			<arg>$DUMPFILE$</arg>
+			<input>!dumpmodule packages $javaBaseModuleAddr$</input>
+			<input>quit</input>
+		</command>
+		<saveoutput regex="no" type="success" saveName="nonExportedPackageAddr" splitIndex="1" splitBy="!j9package ">jdk/internal/platform </saveoutput>
+		<saveoutput regex="no" type="required" saveName="javaUtilPackageAddr" splitIndex="1" splitBy="!j9package ">java/util </saveoutput>
```

Comment:
Are the trailing spaces (e.g. `>java/util `) actually present? I would hope they're not, in which case this will fail to find the package. We also need to make sure it won't find a subpackage like `java/util/concurrent`.

Reply:
The output of `!dumpmodule package` outputs package name, then a series of spaces for alignment, then the package address, as shown below. Thus, the space catches the end of the package name, preventing sub packages from being caught.
```
//              v Last matched character, a space. This line would match "`moduleA/packageA `"
moduleA/packageA                              !j9package 0x00007FA2D43A9D88
//              v Last matched character, an s. This line would fail (correctly)
moduleA/packageA/subPackage                   !j9package 0x00007FA2D43A9D88
```
Is there something you would like me to change? Presently, it is working like intended.

Reply:
This doesn't directly match a package name: instead it matches the format that DDR outputs the package name in, which is always space padded.

Thus, the space acts as matching the end of the package, rather than a character within the package name.

Reply:
I thought the package name was at the end of a line (which shouldn't have any trailing whitespace). Because it's at the beginning (followed by at least one space), we're good.